R/mockVisitOccurrence.R

```diff
@@ -1,101 +1,96 @@
 #' Function to generate visit occurrence table
 #'
 #' @param cdm the CDM reference into which the  mock visit occurrence table will be added
-#' @param recordPerson The expected number of records per person within each cohort. This can help simulate the frequency of observations for individuals in the cohort.
 #' @param seed A random seed to ensure reproducibility of the generated data.
 #'
-#' @return A cdm reference with the mock tables
-#' @noRd
+#' @return A cdm reference with the visit_occurrence tables added
+#' @export
 #'
 #' @examples
 #' library(omock)
 #'
-#' cdm <- mockCdmReference() |>
-#'   mockPerson() |>
-#'   mockObservationPeriod() |>
-#'   mockVisitOccurrence()
-#'
-#' cdm$visit_occurrence
-#'
 mockVisitOccurrence <- function(cdm,
-                                recordPerson = 1,
-                                seed = 1) {
+                                seed = NULL) {
   checkInput(
     cdm = cdm,
-    recordPerson = recordPerson,
     seed = seed
   )
 
   if (!is.null(seed)) {
     set.seed(seed = seed)
   }
 
-  # check if table are empty
-  if (cdm$person |> nrow() == 0 &&
-      cdm$observation_period |> nrow() == 0 && is.null(cdm$concept) &&
-      cdm$condition_occurrence |> nrow() == 0) {
-    cli::cli_abort("person, observation_period and
-                   condition_occurrence table cannot be empty")
-  }
-
 
+  # check for table with persion_id or vist_occurrence_id
+  tableName <- c()
+  for (tab in names(cdm)) {
+    if (all(c("person_id", "visit_occurrence_id") %in% colnames(cdm[[tab]]))) {
+      tableName <-
+        c(tableName, tab)
+    }
+  }
 
-  concept_id <-
-    cdm$concept |>
-    dplyr::filter(.data$domain_id == "Visit") |>
-    dplyr::select("concept_id") |>
-    dplyr::pull() |>
-    unique()
+  if (length(tableName) == 0) {
+    cli::cli_abort("Your cdm object don't contain clinical tables with visit_occurrence_id.")
```

Comment:
no error, just return the cdm the way it is and throw a warning

Reply:
done